### DISCUSSION TITLE
Way to grab a list of tile co-ords from a layer?

### DISCUSSION THREAD

**Discussion by ValleyDragon888** [littlektframework/littlekt#240] [Q&A]
Is there a way of getting a list of the tile co-ordinates from a `TiledTilesLayer`? The only way I could think to do it is to use the `tiles: Map<Int, TiledTileset.Tile>` property and then convert the Id into co-ordinates but `tiles` is private. Is there another way?

**Comment by LeHaine** [comment]
What's the use case? Are you trying to get tile data (size, offset, etc) for a specific tile at a coordinate? Or just general data of the tileset? Under `TiledMap` there is a public list of tilesets `tileSets` that you can iterate through. Depending on the use case, maybe we can make the `tiles` property public, since it isn't a `MutableMap`.

**Reply by LeHaine** [reply]
Now that I dig deeper, I can see that `tiles` is just `TileSets` mapped to the tile id. You can get the same thing doing `val tiles = tiledMap.tileSets.flatMap { it.tiles }.associateBy { it.id }`

**Reply by ValleyDragon888** [reply]
So, just for clarification, I want to get the co-ordinates of every tile on a `TiledTilesLayer`
As well as looking through the source, I also tried reading through the `.tmj` file but apparently it is not stock json :(

**Reply by LeHaine** [reply]
If you want to iterate through each tile, by tile coordinates, you can easily calculate that by using multiplying the size of the layer with the size of the tiles. 

If you are trying to map a tile type (the id) to a list of coordinates that's a little more involved and doesn't seem that the code has the proper access to get that data. The `tileData` property is private and an array, which can be mutated from an outside source. Perhaps, we can add a read only lazy `List` that converts the `tileData` tile bits list into an immutable list of `TileData`. Currently, that is also `internal`.
